ci-jobs: document temp schema macro limitation (#4187)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/ci-jobs.md

<details>
  <summary>Temporary schemas aren't dropping</summary>
  <div>

>     <div>If your temporary schemas aren't dropping after a PR merges or closes, this typically indicates one of two issues:

Reviewer: ```suggestion
    <div>If your temporary schemas aren't dropping after a PR merges or closes, this typically indicates one of these issues:
```

      <ul>
        <li> You have overridden the <code>generate_schema_name</code> macro and it isn't using <code>dbt_cloud_pr_</code> as the prefix.<br></br><br></br> To resolve this, change your macro so that the temporary PR schema name contains the required prefix. For example:
    <br></br><br></br>

>         ✅ Temporary PR schema name contains the prefix <code>dbt_cloud_pr_</code> (like <code>dbt_cloud_pr_123_456_marketing</code>) <br></br>

Reviewer: ```suggestion
        ✅ Temporary PR schema name contains the prefix <code>dbt_cloud_pr_</code> (like <code>dbt_cloud_pr_123_456_marketing</code>). <br></br>
```

        ❌ Temporary PR schema name doesn't contain the prefix <code>dbt_cloud_pr_</code> (like <code>marketing</code>). <br></br>
        </li>
        <br/>
        <li>

>           A macro is creating a schema, but no dbt models are writing to that schema. dbt Cloud does not drop temporary schemas that were not written to as a result of running a dbt model.

Reviewer: ```suggestion
          A macro is creating a schema but there are no dbt models writing to that schema. dbt Cloud doesn't drop temporary schemas that weren't written to as a result of running a dbt model.
```
